simul/platform/runsimul.go

```diff
@@ -55,11 +61,19 @@ func Simulate(serverAddress, simul, monitorAddress string) error {
 		if err != nil {
 			return err
 		}
-		err = sim.Node(sc)
-		if err != nil {
-			return err
-		}
 		sims[i] = sim
+		func(sc *onet.SimulationConfig) {
+			server.RegisterProcessorFunc(simulInitID, func(env *network.Envelope) {
+				err = sim.Node(sc)
+				if err != nil {
+					log.Error(err)
+				}
+				sc.Server.Send(env.ServerIdentity, &simulInitDone{})
+			})
+			server.RegisterProcessorFunc(simulInitDoneID, func(env *network.Envelope) {
+				wgSimulInit.Done()
+			})
+		}(sc)
```

Comment:
Why do you put that in a func ? You don't call it in a goroutine, neither sc is changing... ?